File: Backend/src/main/java/com/resumebuilder/backend/dto/EducationDTO.java
package com.resumebuilder.backend.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import jakarta.validation.constraints.NotBlank;
import java.time.LocalDate;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class EducationDTO {

    private Long id;

    @NotBlank(message = "Institution is required")
    private String institution;

    @NotBlank(message = "Degree is required")
    private String degree;

    private String fieldOfStudy;
    private LocalDate startDate;
    private LocalDate endDate;
    private Double gpa;
}

File: app/src/main/java/com/snehil/cvoptima/data/remote/model/EducationDto.kt
package com.snehil.cvoptima.data.remote.model

import com.google.gson.annotations.SerializedName

data class EducationDto(
    @SerializedName("id") val id: Long? = null,
    @SerializedName("institution") val institution: String,
    @SerializedName("degree") val degree: String,
    @SerializedName("fieldOfStudy") val fieldOfStudy: String?,
    @SerializedName("startDate") val startDate: String?,
    @SerializedName("endDate") val endDate: String?,
    @SerializedName("gpa") val gpa: Double?
)
